src/components/Comments.jsx

```diff
@@ -0,0 +1,267 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { connect } from 'react-redux';
+import { Input } from 'reactstrap';
+import moment from 'moment';
+import store from '../store/store';
+import {
+  createAComment,
+  createAReply,
+} from '../store/actions/commentsActions';
+import 'bootstrap';
+import 'jquery';
+
+export class Comments extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      thisComment: '',
+      thisReply: '',
+      comments: '',
+      slug: '',
+    };
+  }
+
+  componentDidMount() {
+    const { comments, slug } = this.props.data;
+    console.log(this.props);
+    this.setState({
+      comments,
+      slug,
+    });
+  }
+
+  componentWillReceiveProps(newProps) {
+    const { comments } = this.state;
+    const { Comment } = newProps.newComment;
+    comments.unshift(Comment);
+    this.setState({
+      comments,
+    });
+  }
+
+  handleSubmitComment = (event) => {
+    event.preventDefault();
+    const { thisComment, slug } = this.state;
+    const { createComment } = this.props;
+    createComment(slug, thisComment);
+    this.setState({
+      thisComment: '',
+    });
+  }
+
+  handleChange(event) {
+    console.log(event.target.name);
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  handleSubmitReply() {
+    console.log(this.state.thisReply);
+  }
+
+
+  render() {
+    // const { data } = this.props;
+    const { comments } = this.state;
+    return (
+      <div className="row my-comments">
+        <div className="col-md-10 offset-md-1">
+          {/* Add review section */}
+          <br />
+          <br />
+          <hr />
+          <h2>Comments</h2>
+          <br />
+          <div className="card">
+            <br />
+            <div className="row">
+              <br />
+              <div className="col-sm-12">
+                <form>
+                  <div className="form-row">
+                    <div className="col-md-1">
+                      &nbsp;
+                    </div>
+                    <div className="col-md-8">
+                      <Input
+                        className="form-control mb-4"
+                        placeholder="Leave a comment..."
+                        name="thisComment"
+                        type="textarea"
+                        onChange={this.handleChange}
+                      />
+                    </div>
+                    <div className="col-md-3 centered">
+                      <button
+                        type="button"
+                        onClick={this.handleSubmitComment}
+                        className="btn btn-one submitComment"
+                      >
+                      Comment
+                      </button>
+                      <br />
+                    </div>
+                  </div>
+                </form>
+              </div>
+            </div>
+          </div>
+          {
+          (comments.length > 0) ? (
```

Comment:
@Etomovich, it is not good to place so much JSX in a conditional statement. You could make some of the JSX a component of its own.